### PR TITLE
set response flag for mixer filter

### PR DESCRIPTION
**Pull request by kyessenov**

### CODE REVIEW

File: src/envoy/http/mixer/filter.cc [hunk 1]
```diff
@@ -221,6 +221,8 @@ void Filter::completeCheck(const CheckResponseInfo& info) {
     int status_code = ::istio::utils::StatusHttpCode(status.error_code());
     decoder_callbacks_->sendLocalReply(Code(status_code), status.ToString(),
                                        nullptr);
+    decoder_callbacks_->requestInfo().setResponseFlag(
+        RequestInfo::ResponseFlag::UnauthorizedExternalService);
```
Comment (mandarjog) [inline src/envoy/http/mixer/filter.cc]: Is suppose `fail policy` is used somewhere in `istio::utils::StatusHttpCode` or `sendLocalReply` ?

Comment (kyessenov) [reply src/envoy/http/mixer/filter.cc]: It's in gRPC transport used by mixerfilter. Fail-open means you will get status ok here.

**Comment by mandarjog** [comment]
partially addresses https://github.com/istio/istio/issues/7959

File: src/envoy/http/mixer/filter.cc (hunk 1 above)
Comment (mandarjog) [reply src/envoy/http/mixer/filter.cc]: thnx, can you update the description to `Set ResponseFlag`

**Comment by kyessenov** [comment]
@sebastienvas something is wrong with prow build environment in proxy...

**Pull request merged**